cpp/src/arrow/compute/kernels/aggregate_internal.h

```diff
@@ -52,6 +52,16 @@ struct FindAccumulatorType<I, enable_if_floating_point<I>> {
   using Type = DoubleType;
 };
 
+template <typename I>
+struct FindAccumulatorType<I, enable_if_decimal32<I>> {
+  using Type = Decimal64Type;
```

Comment:
Why not `Decimal32Type`?

Reply:
It doesn't make sense, especially if the decimal precision remains the same (which it probably does?).

Reply:
By the way, the reason we upcast integers when summing them is to emulate what DB engines typically do. I don't think DB engines upcast decimals when summing them.

Reply:
That's a good point. we aren't changing the precision when we do this, so if it is necessary to avoid overflow, the type would need to be cast to increase the precision etc. I'll change it back to be the appropriate types